src/main/java/com/achdev/onlinebookstoreapp/repository/querydsl/QPredicates.java

```diff
@@ -0,0 +1,38 @@
+package com.achdev.onlinebookstoreapp.repository.querydsl;
+
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.core.types.dsl.Expressions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class QPredicates {
```

Comment:
It's nice that you find the other way to implement this task, but le't use JPA Specification to implement searching

Reply:
Ok)